main.js

```diff
@@ -0,0 +1,285 @@
+    if(isDown) {    // Only change css if mouse is down
+      $(this).css('background-color', $orange);
+    }
+  });
+  $(".grid1").mouseover(function(){
+    if(isDown) {    // Only change css if mouse is down
+      $(this).css('background-color', $orange);
+    }
+  });
+  $(".grid2").mouseover(function(){
+    if(isDown) {    // Only change css if mouse is down
+      $(this).css('background-color', $orange);
+    }
+  });
+});
+
+// ******************   Green Button  *****************
+
+$('#button3').on('click', function(){
+  var $green = $(this).css('background-color');
+
+  $(".grid").mouseover(function(){
+    if(isDown) {    // Only change css if mouse is down
+      $(this).css('background-color', $green);
+    }
+  });
+  $(".grid1").mouseover(function(){
+    if(isDown) {    // Only change css if mouse is down
+      $(this).css('background-color', $green);
+    }
+  });
+  $(".grid2").mouseover(function(){
+    if(isDown) {    // Only change css if mouse is down
+      $(this).css('background-color', $green);
+    }
+  });
+});
+
+// ******************   yellow Button  *****************
+
+$('#button4').on('click', function(){
+  var $yellow = $(this).css('background-color');
+
+  $(".grid").mouseover(function(){
+    if(isDown) {    // Only change css if mouse is down
+      $(this).css('background-color', $yellow);
+    }
+  });
+  $(".grid1").mouseover(function(){
+    if(isDown) {    // Only change css if mouse is down
+      $(this).css('background-color', $yellow);
+    }
+  });
+  $(".grid2").mouseover(function(){
+    if(isDown) {    // Only change css if mouse is down
+      $(this).css('background-color', $yellow);
+    }
+  });
+});
+
+// ******************   pink Button  *****************
+
+$('#button5').on('click', function(){
+  var $pink = $(this).css('background-color');
+
+  $(".grid").mouseover(function(){
+    if(isDown) {    // Only change css if mouse is down
+      $(this).css('background-color', $pink);
+    }
+  });
+  $(".grid1").mouseover(function(){
+    if(isDown) {    // Only change css if mouse is down
+      $(this).css('background-color', $pink);
+    }
+  });
+  $(".grid2").mouseover(function(){
+    if(isDown) {    // Only change css if mouse is down
+      $(this).css('background-color', $pink);
+    }
+  });
+});
+
+// ******************   Grey Button  *****************
+
+$('#button6').on('click', function(){
+  var $grey = $(this).css('background-color');
+
+  $(".grid").mouseover(function(){
+    if(isDown) {    // Only change css if mouse is down
+      $(this).css('background-color', $grey);
+    }
+  });
+  $(".grid1").mouseover(function(){
+    if(isDown) {    // Only change css if mouse is down
+      $(this).css('background-color', $grey);
+    }
+  });
+  $(".grid2").mouseover(function(){
+    if(isDown) {    // Only change css if mouse is down
+      $(this).css('background-color', $grey);
+    }
+  });
+});
+
+// ******************   Black Button  *****************
```

Comment:
all the code for these buttons is basically the exact same. This should be encapsulated into a single function that generates them (like a `createButton` function). Then create something like a `createButtons` that takes an array with the dynamic values and loop over it calling the `createButton` fn each time. DRY 🔥